pyscript.core/docs/README.md

```diff
@@ -0,0 +1,481 @@
+    print(shared)
+</script>
+
+<!-- dedicated interpreter -->
+<script type="micropython" env="my-project-env">
+    # throws an error - shared doesn't exist
+    print(shared)
+</script>
+```
+
+ℹ️ - Please note if the interpreter takes 1 second to bootstrap, multiple *environments* will take *that* second multiplied by the number of different environments, which is why this feature is considered for **advanced** use cases only and it should be discouraged as generic practice.
+
+  </div>
+</details>
+
+
+## Bootstrapping core
+
+In order to have anything working at all in our pages, we need to at least bootstrap *@pyscript/core* functionalities, otherwise all examples and scripts mentioned in this document would just sit there ... sadly ignored by every browser:
+
+```html
+<!doctype html>
+<html>
+    <head>
+        <!-- this is a way to automatically bootstrap @pyscript/core -->
+        <script type="module" src="https://esm.run/@pyscript/core"></script>
+    </head>
+    <body>
+        <script type="micropython">
+            from js import document
+            document.body.textContent = '@pyscript/core'
+        </script>
+    </body>
+</html>
+```
+
+As *core* exposes some utility/API, using the following method would also work:
+
+```html
+<script type="module">
+    import {
+        define,      // define a custom type="..."
+        whenDefined, // wait for a custom type to be defined
+        XWorker      // allows JS <-> Interpreter communication
+    } from 'https://esm.run/@pyscript/core';
+</script>
+```
+
+Please keep reading this document to understand how to use those utilities or how to have other *Pogramming Languages* enabled in your page via `<script>` elements.
+
+
+## How Scripts Work
+
+The [&lt;script&gt; element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script) has at least these extremely important peculiarities compared to any other element defined by the [HTML Standard](https://html.spec.whatwg.org/multipage/):
+
+ * its only purpose is to contain *data blocks*, meaning that browsers will never try to parse its content as generic *HTML* (and browsers will completely ignore either its content or its attributes, including the `src`, when its [type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type) is not known)
+ * its completely unobtrusive when it comes to both *aria* and *layout*, indeed it's one of the few nodes that can be declared almost anywhere without breaking its parent tree (other notable exception would be a comment node)
+ * for our specific use case, it already offers attributes that are historically well understood and known, also simplifying somehow the creation of this document
+
+The long story short is that any `<script type="non-standard-type">` has zero issues with any browser of choice, but it's true that using some specific *custom type* might lead to future issues in case that `type` could have some special meaning for the future of the Web.
```

Comment:
If this is referring to `Custom Elements` and the fact that inline code is broken, I'd be more explicit here...

Reply:
no, it refers to the fact that `<script type="python">` or `<script type="json">` might conflict in the future, there's nothing about *custom elements* in core because no *custom element* in the registry is strictly needed.